### PR TITLE
[QA] Update `tenacity` dependency to fix failing unit tests

### PR DESCRIPTION
**Pull request by Anna-Xiong**
## Description

Fix failing vizro ai unit tests:
Set tenacity upper bound to avoid issue described in https://github.com/langchain-ai/langchain/issues/22972.

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

**Review by maxschulz-COL — APPROVED** [review]
Thanks!

**Comment by antonymilne** [comment]
What is the version that actually causes a problem here? If it's 8.4.0 then a much better version specifier would be `!=8.4.0`, so that when a fixed version, say 8.4.1, comes out then you don't need to change the requirement again.

**Comment by Anna-Xiong** [comment]
> What is the version that actually causes a problem here? If it's 8.4.0 then a much better version specifier would be `!=8.4.0`, so that when a fixed version, say 8.4.1, comes out then you don't need to change the requirement again.

It is the newly released version 8.4.0(  only released a few hours ago), they now have even newer version 8.4.1 i can try to see if it resolves the issue, otherwise probably let's pin under 8.4.0?

File: vizro-ai/snyk/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 pandas
 tabulate
 openai>=1.0.0
+tenacity==8.4.1
```
Comment (huong-li-nguyen) [inline vizro-ai/snyk/requirements.txt:4]: Seems like the latest version seems to be working 🥳  Do you want to apply the suggested change from @antonymilne? Should be good to go then 👍 

```suggestion
tenacity!=8.4.1
```

File: vizro-ai/pyproject.toml [hunk 2]
```diff
@@ -17,6 +17,7 @@ dependencies = [
   "pandas",
   "tabulate",
   "openai>=1.0.0",
+  "tenacity==8.4.1",  # TODO update when tenacity 8.4.0 is compatible with langchain
```
Comment (huong-li-nguyen) [inline vizro-ai/pyproject.toml:20]:
```suggestion
  "tenacity=!=8.4.1",  # https://github.com/langchain-ai/langchain/issues/22972
```

**Review by huong-li-nguyen — APPROVED** [review]

**Review by antonymilne — CHANGES_REQUESTED** [review]
If the tenacity 8.4.1 has been released and fixes the problem then there should be no need for any changes at all here and we shouldn't add a transitive dependency to our requirements. Anyone doing `pip install vizro-ai` will get the latest version of tenacity unless there's some weird conflict that we know of that forces them into the problematic 8.4.0. So best to just not do anything here I think and close the PR.

**Comment by huong-li-nguyen** [comment]
> If the tenacity 8.4.1 has been released and fixes the problem then there should be no need for any changes at all here and we shouldn't add a transitive dependency to our requirements. Anyone doing `pip install vizro-ai` will get the latest version of tenacity unless there's some weird conflict that we know of that forces them into the problematic 8.4.0. So best to just not do anything here I think and close the PR.

Ah yes, main seems to be fixed now as well!

**Pull request closed without merging**